src/CommunityToolkit.Maui/Layouts/StateLayout/StateLayoutController.shared.cs

```diff
@@ -0,0 +1,326 @@
+	/// <param name="animate"></param>
+	public async void SwitchToContent(bool animate)
+	{
+		if (!layoutWeakReference.TryGetTarget(out var layout))
+		{
+			return;
+		}
+
+		var token = RebuildAnimationTokenSource(layout);
+
+		previousState = LayoutState.None;
+		await ChildrenFadeTo(layout, animate, true);
+
+		if (token.IsCancellationRequested)
+		{
+			return;
+		}
+
+		// Put the original content back in.
+		layout.Children.Clear();
+
+		foreach (var item in originalContent)
+		{
+			item.Opacity = animate ? 0 : 1;
+			layout.Children.Add(item);
+		}
+
+		await ChildrenFadeTo(layout, animate, false);
+	}
+
+	/// <summary>
+	/// Display the StateView template for the given custom state key.
+	/// </summary>
+	/// <param name="customState"></param>
+	/// <param name="animate"></param>
+	public void SwitchToTemplate(string customState, bool animate)
+		=> SwitchToTemplate(LayoutState.Custom, customState, animate);
+
+	/// <summary>
+	/// Display the StateView template for the given LayoutState option.
+	/// </summary>
+	/// <param name="state"></param>
+	/// <param name="customState"></param>
+	/// <param name="animate"></param>
+	/// <exception cref="ArgumentException"></exception>
+	public async void SwitchToTemplate(LayoutState state, string? customState, bool animate)
+	{
+		if (!layoutWeakReference.TryGetTarget(out var layout))
+		{
+			return;
+		}
+
+
+		var token = RebuildAnimationTokenSource(layout);
+
+		// Put the original content somewhere where we can restore it.
+		if (previousState == LayoutState.None)
+		{
+			originalContent = new List<View>();
+
+			foreach (var item in layout.Children)
+			{
+				originalContent.Add((View)item);
+			}
+
+		}
+
+		var view = GetViewForState(state, customState);
+
+		if (view != null)
+		{
+			previousState = state;
+
+			await ChildrenFadeTo(layout, animate, true);
+
+			if (token.IsCancellationRequested)
+			{
+				return;
+			}
+
+
+			layout.Children.Clear();
+
+			var repeatCount = GetRepeatCount(state, customState);
+			var template = GetTemplate(state, customState);
+
+			if (template != null)
+			{
+				// We have a template we can use.
+				var items = new List<int>();
+
+				for (var i = 0; i < repeatCount; i++)
+				{
+					items.Add(i);
+				}
+
+				// We create a StackLayout to stack repeating items.
+				// It takes VerticalOptions and HorizontalOptions from the
+				// StateView to allow for more control over how it layouts.
+				var s = new StackLayout
+				{
+					Opacity = animate ? 0 : 1,
+					VerticalOptions = view.VerticalOptions,
+					HorizontalOptions = view.HorizontalOptions
+				};
+
+				// If the layout we're applying StateLayout to is a Grid,
+				// we want to have the StateLayout span the entire Grid surface.
+				// Otherwise it would just end up in row 0 : column 0.
+				if (layout is Grid grid)
+				{
+					if (grid.RowDefinitions.Any())
+					{
+						Grid.SetRowSpan(s, grid.RowDefinitions.Count);
+					}
+
+
+					if (grid.ColumnDefinitions.Any())
```

Comment:
We don't need to use LINQ here.
```suggestion
					if (grid.ColumnDefinitions.Count > 0)
```